### ISSUE TITLE
Practical usage of embeddings for face recognition

### ISSUE THREAD

**Issue by FSet89** [omoindrot/tensorflow-triplet-loss#26]
I have a training set with N different identities [I_1, I_2 ... I_N] and for each identity I have several images extracted from a video. I trained the triplet loss model on this set. Now I have another set of images related to a new identity I_N+1 and I want all the future images related to I_N+1 to be recognized. What I thought is:
- Compute the mean embedding of the available images related to I_N+1
- When a new image is provided (i.e. at serving time), compute the embedding and compare it with the mean embedding. If the difference is below a given threshold, the identity is confirmed.

Is this approach correct? Is it better to compute several embeddings related to I_N+1 and use them separately?

**Comment by omoindrot** [comment]
I'm not sure if the mean embedding will work here.
Maybe the person has very different poses, different ages or different hair colors across photos. So you might want to compare each embedding in your history to the new unknown embedding.

The only way to know for sure is to test your ideas on a separate test set with multiple identities unseen during training.
